community/cloud-foundation/templates/healthcheck/healthcheck.py

```
@@ -0,0 +1,140 @@
+            },
+        'TCP':
+            {
+                'v1': 'compute.v1.healthCheck',
+                'beta': 'compute.beta.healthCheck'
+            },
+        'HTTP2': {
+            'beta': 'compute.beta.healthCheck'
+        }
+    }
+
+    # Deployment Manager object types associated with each type of healthcheck.
+    healthcheck_object_dictionary = {
+        'HTTP': 'httpHealthCheck',
+        'HTTPS': 'httpsHealthCheck',
+        'SSL': 'sslHealthCheck',
+        'TCP': 'tcpHealthCheck',
+        'HTTP2': 'http2HealthCheck'
+    }
+
+    #Create a generic healthcheck object.
+    healthcheck = {
+        'name':
+            healthcheck_name,
+        'type':
+            healthcheck_type_dictionary[healthcheck_type][healthcheck_version]
+    }
+
+    #Create the generic healthcheck properties separately.
+    healthcheck_properties = {
+        'description': properties.get('description',
+                                      ''),
+        'checkIntervalSec': properties.get('checkIntervalSec'),
+        'timeoutSec': properties.get('timeoutSec'),
+        'unhealthyThreshold': properties.get('unhealthyThreshold'),
+        'healthyThreshold': properties.get('healthyThreshold'),
+        'kind': 'compute#healthCheck',
+        'type': healthcheck_type
+    }
+
+    #Create a specific healthcheck object.
+    specific_healthcheck_type = healthcheck_object_dictionary[healthcheck_type]
+    specific_healthcheck = {
+        'proxyHeader': properties.get('proxyHeader',
+                                      'NONE'),
+    }
+
+    #Check for beta-specific properties.
+    #Add them to the specific healthcheck object.
+    if healthcheck_version == 'beta':
+        if 'portSpecification' in properties:
+            specific_healthcheck['portSpecification'
+                                ] = properties.get('portSpecification')
+        if 'portName' in properties:
+            specific_healthcheck['portName'] = properties.get('portName')
```

Not an error, but as you already checked for existence of `portName` in dict, `properties.get` can be safely replaced with `properties[]`. In a long run it will fail faster in case of typos in property names and therefore will save you debugging time.

---

Can you follow up with this one? I don't think I've checked for portname before this point in the code